spec-0008/index.md

```diff
@@ -0,0 +1,227 @@
+# SPEC-8: Securing the Release Process
+
+---
+
+title: "SPEC 8 — Securing the Release Process"
+number: 8
+date: 2024-06-04
+author:
+
+- "Matthew Feickert <matthew.feickert@cern.ch>"
+- "Pamphile Roy <roy.pamphile@gmail.com>"
+- "Juanita Gomez <juanitagomezr2112@gmail.com>"
+- "Seth Larson <sethmichaellarson@gmail.com>"
+- "Lars Grüter <lagru@mailbox.org>"
+  discussion: https://discuss.scientific-python.org/t/spec-8-supply-chain-security
+  endorsed-by:
+
+---
+
+## Description
+
+<!--
+Briefly and clearly describe the recommendation.
+https://github.com/scientific-python/summit-2024/issues/9
+-->
+
+Securing the Open Source supply chain (OSSC) is becoming a more relevant concern in recent years, with examples of sophisticated attacks against the ecosystem (e.g., the 2024 [`xz` utils backdoor](https://en.wikipedia.org/wiki/XZ_Utils_backdoor)) and [malware attacks on PyPI](https://blog.pypi.org/posts/2024-04-10-domain-abuse/) highlighting the need for supply chain security to be taken seriously.
+The Python Software Foundation (PSF) is also taking the importance of the OSSC seriously, as demonstrated by the [creation of the PSF Security Developer in Residence position in 2023](https://pyfound.blogspot.com/2023/06/announcing-our-new-security-developer.html).
+
+With the creation of the [Supply-chain Levels for Software Artifacts (SLSA) framework](https://slsa.dev/) and [OpenID Connect (OIDC)](https://openid.net/developers/how-connect-works/) standard being more widely adopted there are now high level developer tools, maintained by professional security teams, that are available with clear recommendations on how to use them.
+
+This SPEC outlines the requirements for practically adopting these security tools and recommendations with a focus on securely publishing release artifacts.
+Securely building release artifacts will be covered in a later SPEC. This set of recommendations complements the recommendations from [SPEC 6 — Keys to the Castle](https://github.com/scientific-python/specs/blob/main/spec-0006/index.md).
+
+This SPEC is written with GitHub in mind, note these recommendations still apply with other services such as GitLab and we refer project to these services' documentations and guidelines.
+
+## Implementation
+
+With a focus on securing the release artifact distribution process, the following processes and standards should be adopted.
+
+### Document the release process
+
+The release process should be clearly and fully documented in the developer documentation and describe each step to make a release and the permissions required to do so.
```

Comment:
Maybe give a few example steps, important steps that should be documented. To give people an idea of what steps are involved and at what level of detail they should describe it.

In particular when I compare this section (where people have to do something) to the intro section it feels like a lot of words are spent on the intro and too few on describing what to do.

Reply:
The most important thing is that it is fully described somewhere and does not leave anything out. i.e. anyone with the correct permissions (depends on the tasks you have to do) should just have to strictly follow the step-by-step and arrive to a release. If there is anything not described, the document is incomplete. I am not sure what to add here besides being more strict in the wording as I am now.

Reply:
Maybe give a few links to existing ones that people can take inspiration from.

Reply:
Maybe @matthewfeickert you have one?

I don't know of a good example to be honest and since I am not a release manager (of a large project), I cannot even improve one as I would not know what this person does for the project.

Reply:
One suggestion is to document the release process in a `RELEASING.md`, this is a fairly popular way: 

* https://github.com/search?q=path%3ARELEASING.md&type=code

Here's some examples from projects I'm involved in:

* https://github.com/python-pillow/Pillow/blob/main/RELEASING.md
* https://github.com/hugovk/norwegianblue/blob/main/RELEASING.md
* https://github.com/python/cherry-picker/blob/main/RELEASING.md
* https://github.com/jazzband/tablib/blob/master/RELEASING.md

I find it useful to have a known file to look for when I'm making a release.

Reply:
My suggestion is about the _existence_ of a file like `RELEASING.md` in the repo to document the release process, not to suggest the _contents_.

The links are some examples, and they're quite different from each other. I don't suggest any of the actual contents of those links here. It's up to each project to document whatever their own release process is, which will vary based on their own setup, requirements, and levels of automation etc.

By writing down the steps, it makes it easier for the next maintainer to make a release, and identify new parts which can be automated, and helps enable iterative improvement (this [sort](https://www.transposit.com/devops-blog/devops/2020.04.02-documentation-as-a-path-to-devops-automation/) of stuff).

Anyway, feel free to disregard this suggestion :)

Reply:
Thanks for pointing that out @hugovk and also sorry for not reading carefully enough before &mdash; not enough sleep this week it seems. Your comments make sense to me. I've noted that `RELEASING.md` is a common approach in the latest commit.